submissions/Siusarna/data-structures/data-structures/linkedList.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-underscore-dangle */
+
+const Node = require('./Element.js');
+
+module.exports = class LinkedList {
+  constructor() {
+    this._head = null;
+    this._size = 0;
+  }
+
+  insert(inputValue, successor = null) {
+    if (this._head === null) {
+      this._head = new Node(inputValue);
+      this._size += 1;
+    } else {
+      let iteratorNode = this._head;
+      while (iteratorNode.getNext() && iteratorNode.getValue() !== successor) {
+        iteratorNode = iteratorNode.getNext();
+      }
+      if (successor && iteratorNode.getValue() !== successor) {
+        throw new Error('not found successor');
+      }
+      const prevForInsertedNode = iteratorNode;
+      const currentNode = new Node(inputValue, prevForInsertedNode.getNext());
+      prevForInsertedNode.setNext(currentNode);
+      this._size += 1;
+    }
+  }
+
+  remove(inputValue) {
+    if (this._head.getValue() === inputValue) {
+      this._head = this._head.getNext();
+      this._size -= 1;
+    } else {
+      let iteratorNode = this._head;
+      while (
```

Comment:
This loop looks like loop from 'insert' method. Maybe you can generalize this part of logic?